### DISCUSSION TITLE
path from directory

### DISCUSSION THREAD

**Discussion by omkarprissoft** [cifkao/html-midi-player#16] [Q&A] [answered]
sir how I gave midi file path from my local directory

**Comment by cifkao** [comment, marked as answer]
The player unfortunately cannot play local files, they have to be loaded from a server. This is because JavaScript cannot access local files in modern browsers.

For local development, you need to use an HTTP server, e.g. `http-server` or `live-server` from NPM. Or you can upload your file somewhere.

**Comment by sjonany** [comment]
Just wanted to write down a more concrete list of steps:
- I uploaded my midi to `midi/zanarkand1.mid`
- Then the code looks like so: `<midi-player src="midi/zanarkand1.mid" ...>`
- I use [vscode-preview-server](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/Tools_and_setup/set_up_a_local_testing_server#using_an_extension_in_your_code_editor), and the opened web page would have the plugin working against the midi specified with relative path
